exchanges/credentials.go

```diff
@@ -115,31 +115,29 @@ func (b *Base) GetCredentials(ctx context.Context) (*account.Credentials, error)
 	if value != nil {
 		ctxCredStore, ok := value.(*account.ContextCredentialsStore)
 		if !ok {
-			// NOTE: Return empty credentials on error to limit panic on
-			// websocket handling.
-			return &account.Credentials{}, errContextCredentialsFailure
+			return nil, common.GetTypeAssertError("*account.ContextCredentialsStore", value)
 		}
 
 		creds := ctxCredStore.Get()
 		if err := b.CheckCredentials(creds, true); err != nil {
-			return creds, fmt.Errorf("context credentials issue: %w", err)
+			return nil, fmt.Errorf("context credentials issue: %w", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("error checking context credentials: %w", err)
```

Reply:
🐒 Optional Nitpick
Same as thrasher, but I think that "context credentials" is unnecessarily vague in the error, even though the term is somewhat prolific.
I think when we're throwing blame about it, we can be clearer.
If accepted, this applies to other errors nearby too.
```suggestion
			return nil, fmt.Errorf("error checking credentials from context: %w", err)
```